pkg/controller/controller.go

```diff
@@ -409,9 +413,116 @@ func (c *Controller) handleClusterUpdate(key string) error {
 		}
 	}
 
+	// At this point we have the desired number of statefulsets, and every pod
+	// across those sets is bootstrapped. However some may be bootstrapped because
+	// they own no shards. Check to see that all pods are in the placement.
+	selector := labels.SelectorFromSet(k8sops.GenerateBaseLabels(cluster))
+	pods, err := c.podLister.Pods(cluster.Namespace).List(selector)
+	if err != nil {
+		return fmt.Errorf("error listing pods: %v", err)
+	}
+
+	// TODO(schallert): per-cluster m3admin clients
+	placement, err := c.placementClient.Get()
+	if err != nil {
+		return fmt.Errorf("error fetching active placement: %v", err)
+	}
+
+	c.logger.Info("found placement", zap.Int("currentPods", len(pods)), zap.Int("placementInsts", placement.NumInstances()))
+
+	if placement.NumInstances() < len(pods) {
```

Comment:
Curious if there should be any defense against add too many nodes? If a large amount of hosts are added in quick succession and start peer bootstrapping wouldn't that have an adverse effect on the cluster itself?

Reply:
I think we're protected since as soon as we try to add the second node the coordinator will throw an error since there's currently shards in not-AVAILABLE state

Reply:
Got it, like we discussed! Awesome that works for me.